homeassistant/components/devolo_home_network/config_flow.py

```diff
@@ -108,9 +120,19 @@ async def async_step_zeroconf_confirm(
         if user_input is not None:
             data = {
                 CONF_IP_ADDRESS: self.host,
-                CONF_PASSWORD: "",
+                CONF_PASSWORD: user_input.get(CONF_PASSWORD, ""),
             }
-            return self.async_create_entry(title=title, data=data)
+            try:
+                await validate_input(self.hass, data)
+            except DevicePasswordProtected:
+                return self.async_show_form(
+                    step_id="zeroconf_confirm",
+                    data_schema=STEP_REAUTH_DATA_SCHEMA,
+                    description_placeholders={"host_name": title},
+                    errors={"base": "invalid_auth"},
+                )
```

Comment:
I'm not a big fan of having to `async_show_form` in one step. On error we should set the error dict (that otherwise should be initialized but empty in this step) and update the schema - or even better, can we in zerconf already test connect upon discovery to check whether the user needs to enter a password and then present that as a required field from the start?